src/backend/llvm_ast.c

```diff
@@ -432,6 +432,14 @@ struct LLVMOpaqueModule *blvm_create_module(
     if (compiler_args_print_backend_debug(ctx->args)) {
         bllvm_mod_debug(ctx->llvm_mod, mod_name);
     }
+
+    size_t mod_id_len;
+    char *mod_id = LLVMGetModuleIdentifier(ctx->llvm_mod, &mod_id_len);
```

Comment:
@axic yes I know exactly why.

**TL;DR:**
So ... without going into too many details you will have to move `RFS_POP();` from line `382` to both the happy end and the unhappy end of the function. So move it to line 442 and to line 446.

**More details**: 
`RFstring`, the string we are using from the `rfbase` library is not null terminated. It was designed like that. `LLVM` needs null terminated strings. `rf_string_cstr_from_buff_or_die()` is one of the many functions that create temporary strings by utilizing a temporary buffer that is initialized in the beginning of the program. 

It saves the hustle of allocating and freeing strings but has the drawback that you have to remember where you initialized it with an `RFS_PUSH();` and when you want to stop using it use an `RFS_POP();`

In the current code the `RFS_POP();` is right after the LLVM module initialization call so after that point any other usage of the RFS buffer would overwrite it. Which happens somewhere down the line with the `function_end` label.